### PR TITLE
Upgrade to new Spezi and SpeziFoundation frameworks

### PR DESCRIPTION
**Pull request by Supereg**
# Upgrade to new Spezi and SpeziFoundation frameworks

## :recycle: Current situation & Problem
This PR updates SpeziAccount to the latest changes of the `Spezi` and `SpeziFoundation` frameworks. Additionally, we migrate away from ObservableObject to the new Observable framework.

## :gear: Release Notes 
 
* Upgrade Spezi to `0.8.0`
* Move to `@Observable`.

## :books: Documentation
--

## :white_check_mark: Testing
--
## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Supereg** [comment]
@PSchmiedmayer once approved you may merge the PR. Diff is off due to changes in the Previews.

File: Sources/SpeziAccount/AccountSetupViewStyle/AccountSetupViewStyle.swift [hunk 1]
```diff
@@ -26,6 +26,16 @@ public protocol AccountSetupViewStyle {
...
     var service: Service { get }
 
+    /// A `ViewModifier` that is injected into views with security related operations.
+    ///
+    /// This modifier is injected into views that expose security related operations like changing the password
+    /// or change the user identifier. It is guaranteed that this modifier is not injected twice into the same
+    /// view hierarchy.
+    ///
+    /// - Note: It is advised to implement this as a computed property.
+    var securityRelatedViewModifier: any ViewModifier { get }
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccount/AccountSetupViewStyle/AccountSetupViewStyle.swift:36]: Nice mechanism!

File: Sources/SpeziAccount/SpeziAccount.docc/Account Services/Creating your own Account Service.md [hunk 2]
```diff
@@ -81,13 +81,13 @@ The first approach is to provide your ``AccountService`` as is and let the user
...
-your own `Spezi` [Component](https://swiftpackageindex.com/stanfordspezi/spezi/documentation/spezi/component) that _provides_ your
-``AccountService`` directly to the ``AccountServiceConfiguration`` component. To do so, declare the `@Provide` property wrapper with a type of
+If your account service requires additional setup or any infrastructure that relies on other `Module`s you can implement
+your own `Spezi` [Module](https://swiftpackageindex.com/stanfordspezi/spezi/documentation/spezi/module) that _provides_ your
+``AccountService`` directly to the ``AccountServiceConfiguration`` module. To do so, declare the `@Provide` property wrapper with a type of
 `any AccountService` and populate it within your initializer. Below is a code example.
 
 ```swift
-class MyComponent: Component {
+class MyModule: Module {
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccount/SpeziAccount.docc/Account Services/Creating your own Account Service.md:90]: Thanks for the updates here!

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the changes here, this looks great! 🚀

**Pull request merged**